labgrid/pytestplugin/reporter.py

```diff
@@ -128,29 +128,57 @@ def pytest_runtest_logreport(self, report):
 
 class ColoredStepReporter(StepReporter):
     EVENT_COLORS_DARK = {
+        'expect$': 'blue', # (a lot of output, blend into background)
+        'run': 'magenta',
+        'state_': 'cyan',
+        'transition$': 'yellow',
+        'cycle$|on$|off$': 'white',
+    }
+
+    EVENT_COLORS_LIGHT = {
+        'expect$': 'white', # (a lot of output, blend into background)
+        'run': 'magenta',
+        'state_': 'cyan',
+        'transition$': 'yellow',
+        'cycle$|on$|off$': 'blue',
+    }
+
+    EVENT_COLORS_DARK_256COLOR = {
         'expect$': 8, # dark gray (a lot of output, blend into background)
-        'run': 10, # green
+        'run': 129, # light purple
         'state_': 51, # light blue
         'transition$': 45, # blue
         'cycle$|on$|off$': 246, # light gray
     }
 
-    EVENT_COLORS_LIGHT = {
+    EVENT_COLORS_LIGHT_256COLOR = {
         'expect$': 250, # light gray (a lot of output, blend into background)
-        'run': 10, # green
+        'run': 93, # purple
         'state_': 51, # light blue
         'transition$': 45, # blue
         'cycle$|on$|off$': 8, # dark gray
     }
 
+    EVENT_COLOR_SCHEMES = {
+        'dark': EVENT_COLORS_DARK,
+        'light': EVENT_COLORS_LIGHT,
+        'dark-256color': EVENT_COLORS_DARK_256COLOR,
+        'light-256color': EVENT_COLORS_LIGHT_256COLOR,
+    }
+
     def __init__(self, terminalreporter, *, rewrite=False):
         super().__init__(terminalreporter, rewrite=rewrite)
 
-        scheme = os.environ.get('LG_COLOR_SCHEME', 'dark')
-        if scheme == 'light':
-            self.color_scheme = ColoredStepReporter.EVENT_COLORS_LIGHT
-        else:
-            self.color_scheme = ColoredStepReporter.EVENT_COLORS_DARK
+        default_scheme = 'dark'
+        if os.environ.get('TERM', '').endswith("-256color"):
```

Comment:
This is hugely inaccurate, please use something like the `infocmp` command to retrieve the supported colors.

Reply:
Alright, `curses.tigetnum()` seems to do the trick.